### PR TITLE
Full Backend Structure (Adapted from NFT Maker) + Auth flow

### PR DESCRIPTION
**Pull request by anasnadifiA**

### CODE REVIEW

**Comment by anasnadifiA** [comment]
@ahmadNear @FullStack-Manmohan you may notice seemingly unnecessary files from NFT Maker app, please ignore them for now, since the app conceptualization is not yet finished, we do not know which we may need and which we may not, keeping the files there can then only benefit us, at least for the mean time.

File: app/src/main/java/com/nft/signdoc/repository/SettingsRepository.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+//        {
+//            api.modifyUser(sharePrefs.userId, dToUser)
+//        }
+//        else
+//        {
+//            //send OTP Request then change
+//            val request = DtoLoginRequest(sharePrefs.walletName)
+//            api.login(request)
+//        }
+    }
```
Comment (ahmadNear) [inline app/src/main/java/com/nft/signdoc/repository/SettingsRepository.kt:50]: any specific reason for having this commented code?

Comment (anasnadifiA) [reply app/src/main/java/com/nft/signdoc/repository/SettingsRepository.kt:50]: will be used later, this is a feature to be added in later stage of the project so i thought to keep it there for now, if the feature wont be needed for the project we can later remove it.

File: app/src/main/java/com/nft/signdoc/ui/auth/CreateAccActivity.kt [hunk 2]
```diff
@@ -1,28 +1,144 @@
...
 
+        binding.emailBtn.setOnClickListener {
+            binding.EmailPhone.text?.clear()
+            binding.emailBtn.backgroundTintList = ColorStateList.valueOf(resources.getColor(R.color.selectedColor))
+            binding.phoneBtn.backgroundTintList = ColorStateList.valueOf(Color.WHITE)
+            binding.EmailPhone.hint = this.getString(R.string.email_example)
+            userViewModel.usesPhone = false
+            binding.EmailPhone.inputType = InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS
+            binding.signUpMethod.text = getString(R.string.email)
+
```
Comment (ahmadNear) [inline app/src/main/java/com/nft/signdoc/ui/auth/CreateAccActivity.kt:85]: When user click on this button his input will be reset either he is entering his mobile number or email. For example user is entering his email and accidentally click on this button, his email will be reset and user have to enter his email again from the start. 
You should disable this button once user has selected it until user click on the phoneBtn option and than click on this button.

Same goes for phoneBtn.

File: app/src/main/java/com/nft/signdoc/ui/auth/SignupActivity.kt [hunk 3]
```diff
@@ -20,15 +41,140 @@ class SignupActivity : BaseActivity() {
...
+        initViews()
         listenToViewEvents()
     }
 
+    private fun initViews(){
+        binding.termsText.makeLinks(Pair("Terms & Conditions", View.OnClickListener {
+            val browserIntent = Intent(Intent.ACTION_VIEW)
+            browserIntent.data = Uri.parse("https://terms.nftmakerapp.io/")
+            this.startActivity(browserIntent)
+        }), Pair("Privacy Policy", View.OnClickListener {
```
Comment (ahmadNear) [inline app/src/main/java/com/nft/signdoc/ui/auth/SignupActivity.kt]: Same code is also there in createAccActivity file. Why not  create a global function in Util class and just call that function from both places.

**Review by ahmadNear — COMMENTED** [review]
i have reviewed the PR. It seems fine, most of the code will be use in futurez. There is just couple of points needed your attention which i have mention in the comment.

File: app/src/main/java/com/nft/signdoc/ui/auth/CreateAccActivity.kt (hunk 2 above)
Comment (anasnadifiA) [reply app/src/main/java/com/nft/signdoc/ui/auth/CreateAccActivity.kt:85]: actually this behavior is not my decision, its the same behavior as NFT Maker app, ofcourse that decision came from QA and then from upper management so yeah

File: app/src/main/java/com/nft/signdoc/ui/auth/SignupActivity.kt (hunk 3 above)
Comment (anasnadifiA) [reply app/src/main/java/com/nft/signdoc/ui/auth/SignupActivity.kt]: am taking this into consideration and repushing

**Pull request merged**